pkg/kubelet/status/status_manager.go

```diff
@@ -899,12 +899,12 @@ func (m *manager) canBeDeleted(pod *v1.Pod, status v1.PodStatus, podIsFinished b
 	}
 	// Delay deletion of pods until the phase is terminal.
 	if !podutil.IsPodPhaseTerminal(pod.Status.Phase) {
-		klog.V(3).InfoS("Delaying pod deletion as the phase is non-terminal", "phase", status.Phase, "pod", klog.KObj(pod), "podUID", pod.UID)
+		klog.V(3).InfoS("Delaying pod deletion as the phase is non-terminal", "phase", pod.Status.Phase, "cachedPhase", status.Phase, "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
It might be worth to add a godoc comment to this method to explain the difference between pod.Status and status (and where each ones comes from). i.e. pod.Status is coming from the pod manager (so desired state), while status is the kubelet local status.

Reply:
I have inlined the comments next to the check and the log.